### PR TITLE
IDENTITY-3253

### PR DESCRIPTION
**Pull request by DMHP**
IDENTITY-3253

### CODE REVIEW

File: components/application-authenticators/org.wso2.carbon.identity.application.authenticator.social/src/main/java/org/wso2/carbon/identity/application/authenticator/social/google/GoogleOAuth2Authenticator.java [hunk 1]
```diff
@@ -207,25 +208,27 @@ protected String getAuthenticateUser(OAuthClientResponse token) {
         Map<ClaimMapping, String> claims = new HashMap<ClaimMapping, String>();
 
         try {
-            String json = sendRequest(token.getParam(GoogleOAuth2AuthenticationConstant.GOOGLE_USERINFO_ENDPOINT),
-                    token.getParam(OIDCAuthenticatorConstants.ACCESS_TOKEN));
 
-            Map<String, Object> jsonObject = JSONUtils.parseJSON(json);
+                String json = sendRequest(token.getParam(GoogleOAuth2AuthenticationConstant.GOOGLE_USERINFO_ENDPOINT),
+                        token.getParam(OIDCAuthenticatorConstants.ACCESS_TOKEN));
+            if (json.length() > 0) {
```
Comment (johannnallathamby) [inline components/application-authenticators/org.wso2.carbon.identity.application.authenticator.social/src/main/java/org/wso2/carbon/identity/application/authenticator/social/google/GoogleOAuth2Authenticator.java]: Why not use StringUtils.isNotBlank() ?

File: components/application-authenticators/org.wso2.carbon.identity.application.authenticator.social/src/main/java/org/wso2/carbon/identity/application/authenticator/social/live/WindowsLiveOAuth2Authenticator.java [hunk 2]
```diff
@@ -126,26 +127,27 @@ protected String getAuthenticateUser(OAuthClientResponse token) {
...
+
+                if (jsonObject != null) {
+                    for (Map.Entry<String, Object> entry : jsonObject.entrySet()) {
+                        String key = getClaimUri(entry.getKey());
+                        String value = entry.getValue().toString();
+
+                        if (value != null && value.startsWith("{") && value.endsWith("}")) {
+                            Map<String, Object> children = JSONUtils.parseJSON(value);
+                            if (WindowsLiveOAuth2AuthenticatorConstants.EMAIL_ADD_CLAIM_URI.equals(key)) {
+                                value = (String) children.get("account");
```
Comment (hpmtissera) [inline components/application-authenticators/org.wso2.carbon.identity.application.authenticator.social/src/main/java/org/wso2/carbon/identity/application/authenticator/social/live/WindowsLiveOAuth2Authenticator.java]: define constants

File: components/application-authenticators/org.wso2.carbon.identity.application.authenticator.social/src/main/java/org/wso2/carbon/identity/application/authenticator/social/live/WindowsLiveOAuth2Authenticator.java [hunk 3]
```diff
@@ -126,26 +127,27 @@ protected String getAuthenticateUser(OAuthClientResponse token) {
...
+                            }
                         }
-                    }
 
-                    if (key != null && value != null && value.trim().length() > 0) {
-                        claims.put(ClaimMapping.build(key, key, null, false), value);
-                    }
-                    if (log.isDebugEnabled()) {
-                        log.debug("Adding claim mapping : " + key + " <> " + key + " : " + value);
+                        if (key != null && value != null && value.trim().length() > 0) {
```
Comment (pulasthi7) [inline components/application-authenticators/org.wso2.carbon.identity.application.authenticator.social/src/main/java/org/wso2/carbon/identity/application/authenticator/social/live/WindowsLiveOAuth2Authenticator.java]: can use StringUtils.isNotBlank(value) instead

File: components/application-authenticators/org.wso2.carbon.identity.application.authenticator.social/src/main/java/org/wso2/carbon/identity/application/authenticator/social/live/WindowsLiveOAuth2Authenticator.java (hunk 2 above)
Comment (DMHP) [inline components/application-authenticators/org.wso2.carbon.identity.application.authenticator.social/src/main/java/org/wso2/carbon/identity/application/authenticator/social/live/WindowsLiveOAuth2Authenticator.java]: Defined Constants in pull 935

File: components/application-authenticators/org.wso2.carbon.identity.application.authenticator.social/src/main/java/org/wso2/carbon/identity/application/authenticator/social/live/WindowsLiveOAuth2Authenticator.java (hunk 3 above)
Comment (DMHP) [inline components/application-authenticators/org.wso2.carbon.identity.application.authenticator.social/src/main/java/org/wso2/carbon/identity/application/authenticator/social/live/WindowsLiveOAuth2Authenticator.java]: Done in pull 935

**Pull request merged**